File: ts/src/parsers/AreaChangeParser.ts
import { LogEntryType } from "../enums/LogEntryType";
import { AreaChangeLogEntry } from "../models/entries/AreaChangeLogEntry";
import { LogEntry } from "../models/entries/LogEntry";
import { IParser } from "./abstractions/IParser";

export class AreaChangeParser implements IParser {

	private _regMatch: RegExp = /You have entered .+\\./gi
	private _regsClean: RegExp[] = [
		/: /gi,
		/You have entered /gi,
		/\\./gi
	]

	execute(entry: LogEntry, line: string): LogEntry {
		const cleanedline = this.clean(line);
		return this.parse(cleanedline as string, entry);
	}

	canParse(line: string): boolean {
		return this._regMatch.test(line);
	}

	parse(line: string, entry: LogEntry): AreaChangeLogEntry {
		let areaChangeEntry = new AreaChangeLogEntry(entry);

		areaChangeEntry.types = [LogEntryType.System, LogEntryType.ChangeArea];
		areaChangeEntry.area = line;

		return areaChangeEntry;
	}

	clean(line: string): String {
		this._regsClean.forEach((reg) => line = line.replace(reg, ""));
		return line;
	}

}

File: dotnet/PoeLogsParser/Parsers/AreaChangeParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;
using PoeLogsParser.Enums;
using PoeLogsParser.Models;
using PoeLogsParser.Models.Abstractions;
using PoeLogsParser.Parsers.Abstractions;

namespace PoeLogsParser.Parsers
{
    public class AreaChangeParser : Parser
    {
        private readonly Regex _regMatch = new Regex("You have entered .+\\.", RegexOptions.IgnoreCase);

        private readonly List<Regex> _regsClean = new List<Regex>()
        {
            new Regex(": ", RegexOptions.IgnoreCase),
            new Regex("You have entered ", RegexOptions.IgnoreCase),
            new Regex("\\.", RegexOptions.IgnoreCase)
        };

        public override bool CanParse(string line)
        {
            return _regMatch.IsMatch(line);
        }

        public override ILogEntry Parse(string line, LogEntry entry)
        {
            return new AreaChangeLogEntry(entry) {Types = new[] {LogEntryType.System, LogEntryType.ChangeArea}, Area = line};
        }

        public override string Clean(string line)
        {
            return _regsClean.Aggregate(line, (current, reg) => reg.Replace(current, ""));
        }
    }
}